### PR TITLE
bugfix: performance fix for get_description

### PR DESCRIPTION
**Pull request by matthewhilton**
This fixes a massive performance issue with get_description. Previously would query all courses from the database, and iterates them one by one in php to find the matching course name. This is just plain silly, it should just filter this on the DB.

For example, on a site with 20k courses - this takes 5 seconds.

image

Also I have tweaked the lang strings since they still reference course modules (since this condition was more or less duplicated from the availability_completion code without changing these strings)

Now if a course is missing it shows `missing course` instead of `missing course module`:

Screenshot

### CODE REVIEW

File: classes/condition.php [hunk 1]
```diff
@@ -169,15 +169,14 @@ protected static function get_lang_string_keyword($completionstate) {
...
-      global $DB;
-        // Get name for module.
-        $modc = get_courses();
+        global $DB;
 
-        $modname = get_string('missing', 'availability_othercompleted');
-        foreach ($modc as $modcs) {
-            if($modcs->id == $this->cmid){
-                $modname = $modcs->fullname;
-            }
```
Comment (keevan) [inline classes/condition.php:180]: Ouch. Good improvement you've made to replace this.

Comment (matthewhilton) [reply classes/condition.php:180]: doing this in php seems harder than doing it in the db.... im a bit stumped :upside_down_face:

**Review by keevan — COMMENTED** [review]
As strings are updated, can we get a version bump on this please.

The rest of the code looks like a great improvement to performance :gem: thanks Matt!

**Comment by matthewhilton** [comment]
Good call - have updated version :+1:

**Pull request merged**